exporters/trace/datadog/src/main/java/io/opencensus/exporter/trace/datadog/DatadogExporterHandler.java

```diff
@@ -0,0 +1,191 @@
+import io.opencensus.trace.Tracer;
+import io.opencensus.trace.Tracing;
+import io.opencensus.trace.export.SpanData;
+import io.opencensus.trace.export.SpanExporter;
+import io.opencensus.trace.samplers.Samplers;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+final class DatadogExporterHandler extends SpanExporter.Handler {
+
+  private static final Tracer tracer = Tracing.getTracer();
+  private static final Sampler probabilitySpampler = Samplers.probabilitySampler(0.0001);
+  private static Gson gson =
+      new GsonBuilder()
+          .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
+          .create();
+
+  private final URL agentEndpoint;
+  private final String service;
+  private final String type;
+
+  DatadogExporterHandler(final URL agentEndpoint, final String service, final String type) {
+    this.agentEndpoint = agentEndpoint;
+    this.service = service;
+    this.type = type;
+  }
+
+  @javax.annotation.Nullable
+  private static String attributeValueToString(AttributeValue attributeValue) {
+    return attributeValue.match(
+        Functions.returnToString(),
+        Functions.returnToString(),
+        Functions.returnToString(),
+        Functions.returnToString(),
+        Functions.returnNull());
+  }
+
+  private static Map<String, String> attributesToMeta(
+      final Map<String, AttributeValue> attributes) {
+    final HashMap<String, String> result = new HashMap<>();
+    attributes.forEach((key, value) -> result.put(key, attributeValueToString(value)));
+    return result;
+  }
+
+  private static long convertSpanId(final SpanId spanId) {
+    final byte[] bytes = spanId.getBytes();
+    long result = 0;
+    for (int i = 0; i < Long.SIZE / Byte.SIZE; i++) {
+      result <<= Byte.SIZE;
+      result |= (bytes[i] & 0xff);
+    }
+    if (result < 0) {
+      return -result;
+    }
+    return result;
+  }
+
+  private static long timestampToNanos(final Timestamp timestamp) {
+    return (timestamp == null)
+        ? 0L
+        : TimeUnit.SECONDS.toNanos(timestamp.getSeconds()) + timestamp.getNanos();
+  }
+
+  private static Integer errorCode(final Status status) {
+    if (status == Status.OK || status == Status.ALREADY_EXISTS) {
+      return 0;
+    }
+
+    return 1;
+  }
+
+  String convertToJson(Collection<SpanData> spanDataList) {
+    final ArrayList<DatadogSpan> datadogSpans = new ArrayList<>();
+    for (SpanData sd : spanDataList) {
+      SpanContext sc = sd.getContext();
+
+      final long startTime = timestampToNanos(sd.getStartTimestamp());
+      final long endTime = timestampToNanos(sd.getEndTimestamp());
```

Comment:
This is sort of tricky, the API requires a duration to be sent as part of the span. For now a null endTimestamp is mapped to `0L`, which produces a nonsense value for duration. 

Other options I can think of are to drop those spans or use the current system time, but I don't fully understand when a span would be exported without an endtime set. Do you have an opinion about the right way to handle those?

Reply:
We do export sampled running spans. Since those spans are active, they don't have an end timestamp. For now I'd suggest to use current time as the end timestamp for running spans.